homepage-dashboard: add unixtools.ping to PATH

parthiv-krishna opened the pull request:
This PR adds `unixtools.ping` to the path for the `homepage-dashboard` NixOS package. Homepage added a [`ping` property for services](https://gethomepage.dev/configs/services/#ping) to monitor the status of external hosts. This requires access to `ping`; otherwise, hosts always appear down (see issue #442569). 

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [x] [NixOS tests] in [nixos/tests].
  - [x] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

acid-bong commented on nixos/modules/services/misc/homepage-dashboard.nix:
```diff
@@ -241,6 +241,11 @@ in
 
+      path = with pkgs; [
+        # for `ping` property on services
+        unixtools.ping
```
Ight, `unixtools.ping` is a real attrpath. The title should include `unixtools.ping` or `ping from iputils` tho

parthiv-krishna replied:
Corrected this, thanks for the catch. It's late here oops

acid-bong reviewed (approved):
and good luck nixing

jnsgruk commented:
While this approach works, I wonder if it might be better to adjust the wrapper of the binary so that users of the package *without* the module also benefit from the fix?

The binary used to launch homepage is already wrapped, so  prefixing the path with the binary path of `ping` should be easy enough: https://github.com/NixOS/nixpkgs/blob/8fd2809843014376f8684cd5934673463039c399/pkgs/by-name/ho/homepage-dashboard/package.nix#L92

parthiv-krishna commented:
> I wonder if it might be better to adjust the wrapper of the binary so that users of the package _without_ the module also benefit from the fix?

That's a good call; thanks for the suggestion. I will test this out and make sure it works and add to this PR. 

Also, it seems like the package is now unmaintained (sorry @jnsgruk for tagging you and thanks for helping anyways). I would be happy to pick up maintainer duties for `homepage-dashboard` along with this change. I use homepage frequently and can make sure that it's updated/building.

parthiv-krishna committed:
maintainers: add parthiv-krishna

parthiv-krishna committed:
homepage-dashboard: add `parthiv-krishna` to maintainers

parthiv-krishna committed:
homepage-dashboard: add unixtools.ping to wrapper PATH

This gives homepage access to `ping` so it can be used
when the ping property on a service is configured

parthiv-krishna commented:
Adding to the PATH via the package wrapper seems to work. I am guessing the preferred approach here will be to force-push a clean branch with just the commit editing the package?

jnsgruk commented:
@parthiv-krishna that would be great - I assume you know the process to add yourself as a maintainer? If not let me know and I'll help you and we can get it approved/merged.

I'm not *out out*, so I will probably still be around to answer some questions, but I removed myself because I'm not using the package at the moment, and don't have regular access to a NixOS machine to test properly.

jnsgruk commented:
@ofborg test homepage-dashboard

parthiv-krishna committed:
nixosTests.homepage-dashboard: add test for ping property on service

Use homepage's API to ensure that ping works

jnsgruk reviewed (approved):
LGTM, thanks for your work on this!

jnsgruk commented:
> For future reference, where can I see the results from ofborg?

You'll see the `passthruTests` jobs in the list of Github Actions that run for the PR usually, then you can click on the relevant architecture and you'll see a link to grab the results.
